amundsen_application/static/js/components/TableDetail/ColumnListItem/index.tsx

```diff
@@ -51,15 +52,17 @@ export class ColumnListItem extends React.Component<
   }
 
   toggleExpand = (e) => {
+    const metadata = this.props.data;
     if (!this.state.isExpanded) {
-      const metadata = this.props.data;
       logClick(e, {
         target_id: `column::${metadata.name}`,
         target_type: 'column stats',
         label: `${metadata.name} ${metadata.col_type}`,
       });
     }
-    this.setState({ isExpanded: !this.state.isExpanded });
+    if (this.shouldRenderDescription() || metadata.stats.length !== 0) {
+      this.setState({ isExpanded: !this.state.isExpanded });
```

Comment:
Oh, sorry, I missed this one before.

When we set a state that depends on the previous state, we should always use the functional way:
```
this.setState(prevState => ({
  isExpanded: !prevState.isExpanded
}));
```
This is because `setState` is asynchronous, and it could result on unexpected state being saved.
Check the docs for more info: https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

Reply:
Okay, thank you!